pkg/util/mount/nsenter_mount.go

```diff
@@ -189,9 +189,10 @@ func (n *NsenterMounter) IsLikelyNotMountPoint(file string) (bool, error) {
 		return true, nil
 	}
 	strOut := strings.TrimSuffix(string(out), "\n")
-
 	glog.V(5).Infof("IsLikelyNotMountPoint findmnt output for path %s: %v", file, strOut)
-	if strOut == file {
+
+	//TODO: This is a temperory workaround for fixing the buggy code. Track the solution with issue #26996
+	if strings.Contains(file, strOut) {
```

Comment:
Some clarification about this fix.
1. The missing last letter from the output of findmnt command: I also
tested this command in my local environment with a simple go program. The
result still misses the last letter. Run command line directly on the
terminal has no issue. I also tried adding options "--notruncate", "--raw",
but still not working. So I think the bug is not related to our kubernetes
code.

The output from findmnt is /usr/local/google/home/jinxu/work/src/
k8s.io/jingggggggggggggggggggggggggggggggggggg  //the last letter t is
missing
The file path is /usr/local/google/home/jinxu/work/src/
k8s.io/jinggggggggggggggggggggggggggggggggggggt

2. Use strings.Contains() will not result in false positive. We pass the
testing mountpoint path when executing the findmnt command. As long as it
gives back a result, no matter whether some letters are missing or not, it
means that the given path is a mountpoint. Otherwise, an error returns.

We are thinking to further refactor this code because this function
currently servers as two purposes, one is to test whether the dir exist or
not, the other is testing mountpoint. It maybe be more clean to separate
the functions. Also we are debating whether to use other ways for checking
mountpoint instead of using findmnt

Please let me know if there are some concerns about the fix.

Thanks!
Jing


On Wed, Jun 8, 2016 at 11:45 AM, Tim Hockin <notifications@github.com>
wrote:

> In pkg/util/mount/nsenter_mount.go
> <https://github.com/kubernetes/kubernetes/pull/27054#discussion_r66315282>
> :
>
> >  	glog.V(5).Infof("IsLikelyNotMountPoint findmnt output for path %s: %v", file, strOut)
> > -	if strOut == file {
> > +
> > +	//TODO: This is a temperory workaround for fixing the buggy code. Track the solution with issue #26996
> > +	if strings.Contains(file, strOut) {
>
> I *really* don't like this fix. I do not want it to last very long,
> pleeeeeeease.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/27054/files/602e5aa032deae544e8560635ad3f48d30a7260d#r66315282>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ASSNxWcEZ5i_McdGy-vyTStU0GCaNCDbks5qJw3JgaJpZM4IxMdY>
> .
>



-- 
- Jing